app/features/build_runner/fastlane_build_runner.rb

```diff
@@ -39,6 +39,10 @@ def setup(parameters: nil)
 
     # completion_block is called with an array of artifacts
     def run(new_line_block:, completion_block:)
+      if lane.nil?
+        raise "Before calling `.run` on #{self}, you have to call `.setup` to finish preparing the BuildRunner"
```

Comment:
Is there anything we should do the with the `completion_block`? Does anything rely on it being called?

Reply:
`completion_block` is already a required parameter it seems? Note that we check if `self.lane` was defined, which is set as part of `setup`, so we can ensure the setup was properly executed.

Reply:
Also `run` is called from the parent class only, not from the outside world it seems, coming from calling `start` from the `BuildRunner` object